bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java

```diff
@@ -0,0 +1,371 @@
+                    }
+                    try {
+                        Files.delete(dir);
+                    } catch (IOException ex) {
+                    }
+                    return FileVisitResult.CONTINUE;
+                }
+            });
+        } catch (IOException e) {
+        }
+    }
+
+    private void waitAndClean() {
+        try {
+            // Give max 10 seconds for the server to stop before to delete jbossHome.
+            ModelNode mn = new ModelNode();
+            mn.get(ADDRESS);
+            mn.get(OP).set(READ_ATTRIBUTE_OPERATION);
+            mn.get(NAME).set(SERVER_STATE);
+            for (int i = 0; i < 10; i++) {
+                try {
+                    ModelControllerClient client = server.getModelControllerClient();
+                    if (client != null) {
+                        ModelNode ret = client.execute(mn);
+                        if (ret.hasDefined(RESULT)) {
+                            String val = ret.get(RESULT).asString();
+                            if (STOPPED.equals(val)) {
+                                log.serverStopped();
+                                break;
+                            } else {
+                                log.serverNotStopped();
+                            }
+                        }
+                        Thread.sleep(1000);
+                    } else {
+                        log.nullController();
+                        break;
+                    }
+                } catch (Exception ex) {
+                    log.unexpectedExceptionWhileShuttingDown(ex);
+                }
+            }
+        } finally {
+            cleanup();
+        }
+    }
+
+    /**
+     * Modular entry point.
+     *
+     * @param jbossHome Server home directory.
+     * @param args User provided arguments.
+     * @param moduleLoader JBoss modules loader.
+     * @param moduleClassLoader Bootable jar module classloader
+     * @param unzipTime Time spent to unzip the server.
+     * @throws Exception
+     */
+    public static void run(Path jbossHome, List<String> args, ModuleLoader moduleLoader, ModuleClassLoader moduleClassLoader, Long unzipTime) throws Exception {
+        setTccl(moduleClassLoader);
+        Arguments arguments;
+        try {
+            arguments = Arguments.parseArguments(args);
+        } catch (Throwable ex) {
+            System.err.println(ex);
+            CmdUsage.printUsage(System.out);
+            return;
+        }
+        if (arguments.isHelp()) {
+            CmdUsage.printUsage(System.out);
+            return;
+        }
+        BootableJar bootableJar = new BootableJar(jbossHome, arguments, moduleLoader, unzipTime);
+        bootableJar.run();
+    }
+
+    static void setTccl(final ClassLoader cl) {
+        if (System.getSecurityManager() == null) {
+            Thread.currentThread().setContextClassLoader(cl);
+        } else {
+            AccessController.doPrivileged(new PrivilegedAction<Object>() {
```

Comment:
Is there a reason we're doing this privileged? I ask because if we're running this under a security manager the other places where we set system properties are gonna be an issue as well.

Reply:
That is un-needed, I removed the block.